npm/src/upload/upload-form.ts

```diff
@@ -63,83 +135,117 @@ export class UploadForm {
     })
   }
 
-  addFolderListener() {
-    this.dropzone.addEventListener("drop", (e) => {
-      const droppedObject: File | null = e.dataTransfer!.files[0] // Since we do not allow multiple objects to be dropped, there is always only one file
+  handleDropppedItems: (ev: DragEvent) => any = async (ev) => {
+    const droppedItems: FileList | null = ev.dataTransfer!.files // THIS HAS TO BE A FILELIST
```

Comment:
What does this comment mean? It sounds IMPORTANT but I'm not sure what I'd do with it if I was updating this code.

Reply:
I'll let @techncl have the last word on this one but I don't think we need the comment because the type shows that it's a `FileList`

Reply:
I think (but could be wrong) it's because @suzannehamilton either you were wondering why we were both accessing files via `ev.dataTransfer!.files` and `ev.dataTransfer?.items!` or it's because in an earlier iteration of this, where we were going to create the classes for the DroppedFile and BrowsedFiles, I think you might have suggested a ev.dataTransfer?.items like in the prototype.  I apologise, I should have been clearer; it probably made sense at the time :).

Reply:
Right, so is the comment meant to explain why we use `ev.dataTransfer!.files` in the first half of the method and `ev.dataTransfer?.items!` in the latter half? I agree that that's a bit confusing, so I think it's worth expanding the comment to explain the difference if it's still necessary to use `files` in one place and `items` in the other.

Reply:
Now that I think about it more, I think the reason for it might be due to the latter half of my answer (the implementation I was working on required it). Looking at the documentation, I think that we could actually replace `ev.dataTransfer!.files` with `ev.dataTransfer?.items!` as it also has a `length` property and the items within it have a `type` method. What do you think @MancunianSam?

Reply:
I've managed to get this to work now using only `ev.dataTransfer?.items!`